coffea/util.py

```diff
@@ -105,3 +106,19 @@ def deprecate(exception, version, date=None):
             version, date, type(exception).__name__, str(exception)
         )
         warnings.warn(message, FutureWarning)
+
+
+# re-nest a record array into a ListArray
+def awkward_rewrap(arr, like_what, gfunc):
```

Comment:
A few more awkward private methods to worry about. Maybe we can make an issue for what we want in awkward?
iiuc it is just a way to flatten, apply, and unflatten a record array?